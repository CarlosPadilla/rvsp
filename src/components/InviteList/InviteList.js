import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './InviteListItem';

const InviteList = props => (
  <div>
    {
      props.invites.map(invite => <ListItem key={invite.id} invite={invite} />)
    }
  </div>
);

InviteList.propTypes = {
  invites: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  })).isRequired,
};

export default InviteList;
