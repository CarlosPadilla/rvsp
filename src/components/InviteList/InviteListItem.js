import React from 'react';
import PropTypes from 'prop-types';

const InviteListItem = props => (
  <div>
    <div>{props.invite.id}</div>
    <div>{props.invite.name}</div>
    <div>{props.invite.email}</div>
  </div>
);

InviteListItem.propTypes = {
  invite: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
};

export default InviteListItem;
