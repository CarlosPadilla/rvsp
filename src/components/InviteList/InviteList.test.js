import React from 'react';
import { shallow } from 'enzyme';
import InviteList from './InviteList';

describe('InviteList Component', () => {

  const invites = [
    {
      id: 1,
      name: 'Aimar',
      email: 'aimar@snapit.ee'
    },
    {
      id: 2,
      name: 'Marju',
      email: 'marju@domeca.ee'
    }
  ]

  const app = shallow(<InviteList invites={invites} />)

  it('renders', () => {
    expect(app).toMatchSnapshot()
  })

})
