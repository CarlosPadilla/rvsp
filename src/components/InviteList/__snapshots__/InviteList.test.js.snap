// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteList Component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InviteList
    invites={
        Array [
            Object {
              "email": "aimar@snapit.ee",
              "id": 1,
              "name": "Aimar",
            },
            Object {
              "email": "marju@domeca.ee",
              "id": 2,
              "name": "Marju",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InviteListItem
          invite={
                    Object {
                              "email": "aimar@snapit.ee",
                              "id": 1,
                              "name": "Aimar",
                            }
          }
/>,
        <InviteListItem
          invite={
                    Object {
                              "email": "marju@domeca.ee",
                              "id": 2,
                              "name": "Marju",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "invite": Object {
            "email": "aimar@snapit.ee",
            "id": 1,
            "name": "Aimar",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "invite": Object {
            "email": "marju@domeca.ee",
            "id": 2,
            "name": "Marju",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InviteListItem
            invite={
                        Object {
                                    "email": "aimar@snapit.ee",
                                    "id": 1,
                                    "name": "Aimar",
                                  }
            }
/>,
          <InviteListItem
            invite={
                        Object {
                                    "email": "marju@domeca.ee",
                                    "id": 2,
                                    "name": "Marju",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "invite": Object {
              "email": "aimar@snapit.ee",
              "id": 1,
              "name": "Aimar",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "invite": Object {
              "email": "marju@domeca.ee",
              "id": 2,
              "name": "Marju",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
